.github/workflows/build-yocto.yml

```diff
@@ -29,4 +29,6 @@ jobs:
           export SSTATE_DIR=${PERSISTENT_DIR}/sstate-cache
           mkdir -p $DL_DIR
           mkdir -p $SSTATE_DIR
+          kas dump --update --lock --inplace ci/base.yml
           kas build ci/${{ matrix.machine }}.yml
+          mv ci/base.lock.yml ${PERSISTENT_DIR}
```

Comment:
You run a matrix job, which can potentially run several builds simultaneously. This command here is not 'safe' as multiple runners can write to the same file. And since the kas dump is done several times, it can return something different in case someone pushed a commit in one of the repos.

Reply:
Yes, it's true, so we need some kind of lock between the various builds that run simultaneously.
This seems to be somehow related to https://github.com/quic-yocto/meta-qcom-hwe/issues/37

1. we need a first job to generate a file `ci/base.lock.yml` that will be accessible in the following tasks
2. and we need a last job to save that generated file if all build jobs were successful

Reply:
It depends what we want to do. If you want all N jobs in the matrix to build from the exact same 'lock' file, then we need to generate the lock file first, and then run the matrix job with that lock file. Since the repos we use _can_ change while we run the N matrix jobs. 

and yes, you can add a job that makes the lock that runs first.